### PR TITLE
Port cryptsetup patch from 2.3.2 to 2.3.7

### PR DESCRIPTION
**Pull request by Jasper-Ben**

### CODE REVIEW

**Comment by Jasper-Ben** [comment]
Needs testing before merge.
Build will be found at: https://gitlab.devops.defra01.iris-sensing.net/MIRROR/iris-kas/-/pipelines/2363

Who can take care of testing this? @Eximmius @linkjumper @ThomasWNoack @iris-ersc

**Review by Eximmius — APPROVED** [review]
Tested. Device still boots with CAAM encryption.

File: patches/0001-Port-cryptsetup-patch-to-2.3.7.patch [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+@@ -108,14 +112,14 @@ index 129d64a..9cc7ccb 100644
+  	} else {
+  		key_size_max = (opt_key_file && !params.hash) ? key_size : (size_t)opt_keyfile_size;
+  		r = tools_get_key(NULL, &password, &passwordLen,
+-@@ -3486,6 +3503,7 @@ int main(int argc, const char **argv)
++@@ -3648,6 +3665,7 @@ int main(int argc, const char **argv)
+  		{ "hash",              'h',  POPT_ARG_STRING, &opt_hash,                0, N_("The hash used to create the encryption key from the passphrase"), NULL },
+  		{ "verify-passphrase", 'y',  POPT_ARG_NONE, &opt_verify_passphrase,     0, N_("Verifies the passphrase by asking for it twice"), NULL },
+- 		{ "key-file",          'd',  POPT_ARG_STRING, &opt_key_file,            6, N_("Read the key from a file"), NULL },
++ 		{ "key-file",          'd',  POPT_ARG_STRING, NULL,                     6, N_("Read the key from a file"), NULL },
```
Comment (Jasper-Ben) [inline patches/0001-Port-cryptsetup-patch-to-2.3.7.patch:113]: whoops. Pretty sure this isn't supposed to be NULL :worried:

Comment (Jasper-Ben) [reply patches/0001-Port-cryptsetup-patch-to-2.3.7.patch:113]: nevermind, that's an upstream change. Wrong line :laughing:

**Review by martin31821 — APPROVED** [review]

**Pull request merged**